src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -332,40 +328,40 @@ def post_fork_child(self):
 
     # Invoke a Pants run with stdio redirected and a proxied environment.
     with self.nailgunned_stdio(self._socket, self._env) as finalizer,\
-         hermetic_environment_as(**self._env):
+         hermetic_environment_as(**self._env),\
+         ExceptionSink.exiter_as(self._exiter) as daemon_exiter:
```

Comment:
I believe we could avoid having `.exiter_as()` have to `yield` a value, by instead exposing an accessor to the globally registered exiter. This could look like the edit suggestion I've made in `local_pants_runner.py`. The only reason that might be nicer is because we then wouldn't have to pass the exiter as an argument into `LocalPantsRunner`, it would just read the global value.